src/components/CheckMarkCard/CheckMarkCard.jsx

```diff
@@ -1,178 +1,171 @@
+  return Boolean(tooltip) ? (
+    <Tooltip
+      title={tooltip}
+      triggerOn="mouseenter focus"
+      appendTo={document.body}
+      withTriggerWrapper={false}
+    >
+      <MarkUI className={markClassnames}>{icon}</MarkUI>
+    </Tooltip>
+  ) : (
+    <MarkUI className={markClassnames}>{icon}</MarkUI>
+  )
+}
 
-  getClassName() {
-    const { className, disabled, withStatus } = this.props
-    const { cardChecked } = this.state
-
-    return classNames(
-      'c-CheckMarkCard',
-      className,
-      cardChecked && !Boolean(withStatus) && 'is-checked',
-      disabled && 'is-disabled',
-      Boolean(withStatus) && 'with-status',
-      Boolean(withStatus) &&
-        Boolean(withStatus.status) &&
-        `is-${withStatus.status}`
-    )
+const CheckMarkCard = props => {
+  const {
+    checked,
+    children,
+    className,
+    disabled,
+    label,
+    onBlur,
+    onChange,
+    onFocus,
+    isFocused,
+    inputRef: inputRefProp,
+    maxWidth,
+    height,
+    value: valueProp,
+    withStatus,
+    ...rest
+  } = props
+
+  const checkMarkCardRef = useRef()
+  const inputRef = useRef()
+  const [cardChecked, setCardChecked] = useState(checked)
+  const id = useCustomId(rest.id)
+
+  const handleOnChange = (value, checked) => {
+    setCardChecked(checked)
+    onChange && onChange(value, checked)
   }
 
-  setInputNodeRef = node => {
-    this.inputNode = node
-    this.props.inputRef(node)
+  const handleOnBlur = event => {
+    checkMarkCardRef.current.classList.remove('is-focused')
+    onBlur && onBlur(event)
   }
 
-  handleOnChange = (value, checked) => {
-    this.setState({
-      cardChecked: checked,
-    })
-
-    this.props.onChange(value, checked)
+  const handleOnFocus = event => {
+    checkMarkCardRef.current.classList.add('is-focused')
+    onFocus && onFocus(event)
   }
 
-  handleOnBlur = event => {
-    this.checkMarkCardRef.current.classList.remove('is-focused')
-    this.props.onBlur(event)
+  const setInputNodeRef = node => {
+    inputRef.current = node
+    inputRefProp && inputRefProp(node)
   }
 
-  handleOnFocus = event => {
-    this.checkMarkCardRef.current.classList.add('is-focused')
-    this.props.onFocus(event)
-  }
+  useEffect(() => {
+    setCardChecked(checked)
+  }, [checked])
 
-  renderMark = () => {
-    const { withStatus } = this.props
-    const { cardChecked } = this.state
-    let iconName
-    let iconSize
-    let color
-    let tooltip
-
-    if (cardChecked) {
-      iconName = 'checkmark'
-      iconSize = '24'
-      color = getColor('blue.500')
-    }
-    // If the card has a status provided, it should take precedence even if the card
-    // is checked from external props for some reason
-    if (withStatus) {
-      iconName = withStatus.iconName
-      iconSize = withStatus.iconSize || '20'
-      color = withStatus.color
-      tooltip = withStatus.tooltipText
+  useEffect(() => {
+    // Update the document title using the browser API
```

Comment:
Am I missing where we updating the document title?

Reply:
I bet I copied that useEffect from somewhere else 🤦

I will remove that comment before merging this PR